$MirOS: ports/www/lynx/patches/patch-lynx_cfg,v 1.12 2009/09/06 14:17:03 tg Exp $

	mostly site policy and compatibility, usability

--- lynx.cfg.orig	Tue Jun 23 23:53:34 2009
+++ lynx.cfg	Sun Nov  8 19:26:17 2009
@@ -1,7 +1,8 @@
+# $MirSecuCron$
+# $MirOS: ports/www/lynx/patches/patch-lynx_cfg,v 1.12 2009/09/06 14:17:03 tg Exp $
 # $LynxId: lynx.cfg,v 1.180 2009/06/07 17:02:21 tom Exp $
 # lynx.cfg file.
-# The default placement for this file is /usr/local/lib/lynx.cfg (Unix)
-#                                     or Lynx_Dir:lynx.cfg (VMS)
+# The default placement for this file is /etc/lynx.cfg
 #
 # $Format: "#PRCS LYNX_VERSION \"$ProjectVersion$\""$
 #PRCS LYNX_VERSION "2.8.7rel.1"
@@ -12,7 +13,7 @@
 # Definition pairs are of the form  VARIABLE:DEFINITION
 # NO spaces are allowed between the pair items.
 #
-#  If you do not have write access to /usr/local/lib you may change
+#  If you do not have write access to /etc you may change
 #  the default location of this file in the userdefs.h file and recompile,
 #  or specify its location on the command line with the "-cfg"
 #  command line option.
@@ -93,12 +94,12 @@
 #
 # Normally we expect you will connect to a remote site, e.g., the Lynx starting
 # site:
-STARTFILE:http://lynx.isc.org/
+#STARTFILE:http://lynx.isc.org/
 #
 # As an alternative, you may want to use a local URL.  A good choice for this is
 # the user's home directory:
 .ex
-#STARTFILE:file://localhost/~/
+STARTFILE:file://localhost/~/
 #
 # Your choice of STARTFILE should reflect your site's needs, and be a URL that
 # you can connect to reliably.  Otherwise users will become confused and think
@@ -442,7 +443,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # Lynx (case insensitive).
 # Find RFC 1345 at http://www.ics.uci.edu/pub/ietf/uri/rfc1345.txt .
 #
-#CHARACTER_SET:iso-8859-1
+CHARACTER_SET:utf-8
 
 .h2 LOCALE_CHARSET
 # LOCALE_CHARSET overrides CHARACTER_SET if true, using the current locale to
@@ -467,7 +468,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # ASSUME_CHARSET can also be changed via the 'o'ptions menu but will
 # not be saved as permanent value in user's .lynxrc file to avoid more chaos.
 #
-#ASSUME_CHARSET:iso-8859-1
+ASSUME_CHARSET:utf-8
 
 
 .h2 ASSUMED_DOC_CHARSET_CHOICE
@@ -595,7 +596,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # the 'o'ptions menu and save that preference in their RC file.
 # This may be a comma-separated list of languages in decreasing preference.
 #
-#PREFERRED_LANGUAGE:en
+#PREFERRED_LANGUAGE:
 
 
 .h2 PREFERRED_CHARSET
@@ -614,7 +615,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # the sending of an unacceptable response is also allowed.  See RFC 2068
 # (http://www.ics.uci.edu/pub/ietf/uri/rfc2068.txt).
 #
-#PREFERRED_CHARSET:
+PREFERRED_CHARSET:utf-8
 
 
 .h2 CHARSETS_DIRECTORY
@@ -695,22 +696,22 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # Set the threshold # of lines Lynx must render before it
 # redraws the screen in PARTIAL mode.  Anything < 0 implies
 # use of the screen size.
-#PARTIAL_THRES:-1
+PARTIAL_THRES:3
 
 
 .h2 SHOW_KB_RATE
 # While getting large files, Lynx shows the approximate rate of transfer.
-# Set this to change the units shown.  "Kilobytes" denotes 1024 bytes:
+# Set this to change the units shown.
 #	NONE to disable the display of transfer rate altogether.
-#	TRUE or KB for Kilobytes/second.
+#	TRUE or KB for Kibibytes/second.
 #	FALSE or BYTES for bytes/second.
-#	KB,ETA to show Kilobytes/second with estimated completion time.
-#	BYTES,ETA to show BYTES/second with estimated completion time.
+#	KB,ETA to show Kibibytes/second with estimated completion time.
+#	BYTES,ETA to show bytes/second with estimated completion time.
 # Note that the "ETA" values are available if USE_READPROGRESS was defined.
-#SHOW_KB_RATE:TRUE
+SHOW_KB_RATE:BYTES,ETA
 
 .h2 SHOW_KB_NAME
-# Set the abbreviation for Kilobytes (1024).
+# Set the abbreviation for Kibibytes (1024 Bytes).
 # Quoting from
 #	http://www.romulus2.com/articles/guides/misc/bitsbytes.shtml
 # In December 1998, the International Electrotechnical Commission (IEC)
@@ -720,9 +721,6 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # metric prefixes and adding the first two letters of the word "binary".  Thus,
 # for instance, instead of Kilobyte (KB) or Gigabyte (GB), the new terms would
 # be kibibyte (KiB) or gibibyte (GiB).
-#
-# If you prefer using the conventional (and more common) "KB", modify this
-# setting.
 #SHOW_KB_NAME:KiB
 
 .h1 Timeouts
@@ -859,8 +857,8 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # number of documents to cache (rather than the maximum number only if
 # DEFAULT_VIRTUAL_MEMORY_SIZE has been exceeded, as with VAXC/VAX).
 #
-#DEFAULT_CACHE_SIZE:10
-#DEFAULT_VIRTUAL_MEMORY_SIZE:512000
+DEFAULT_CACHE_SIZE:512
+DEFAULT_VIRTUAL_MEMORY_SIZE:25165824
 
 
 .h2 SOURCE_CACHE
@@ -876,7 +874,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # NONE is the default; the document source is not cached, and is reloaded
 #   from the network when needed.
 #
-#SOURCE_CACHE:NONE
+SOURCE_CACHE:MEMORY
 
 
 .h2 SOURCE_CACHE_FOR_ABORTED
@@ -886,7 +884,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # be preserved (and used as cache), if set to DROP lynx will drop the
 # source cache for that document (i.e. only completely downloaded documents
 # will be cached in that case).
-#SOURCE_CACHE_FOR_ABORTED:DROP
+SOURCE_CACHE_FOR_ABORTED:KEEP
 
 .h2 ALWAYS_RESUBMIT_POSTS
 # If ALWAYS_RESUBMIT_POSTS is set TRUE, Lynx always will resubmit forms
@@ -1124,7 +1122,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # default or that defined here can be toggled via the -force_secure
 # command line switch.
 #
-#FORCE_SSL_COOKIES_SECURE:FALSE
+FORCE_SSL_COOKIES_SECURE:TRUE
 
 
 .h1 Internal Behavior
@@ -1313,7 +1311,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # line break.  Note that the valid way to insert extra blank lines in HTML
 # is via a PRE block with only newlines in the block.
 #
-#COLLAPSE_BR_TAGS:TRUE
+COLLAPSE_BR_TAGS:FALSE
 
 
 .h2 TAGSOUP
@@ -1416,7 +1414,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # The default is FALSE, so that the feature needs to be enabled here
 # explicitly if you want it.
 #
-#PERSISTENT_COOKIES:FALSE
+PERSISTENT_COOKIES:TRUE
 
 
 .h2 COOKIE_FILE
@@ -1425,7 +1423,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # USE_PERSISTENT_COOKIES and the PERSISTENT_COOKIES option is enabled.
 # The cookie file can also be specified in .lynxrc or on the command line.
 #
-#COOKIE_FILE:~/.lynx_cookies
+#COOKIE_FILE:~/.etc/cookies.txt
 
 
 .h2 COOKIE_SAVE_FILE
@@ -1438,7 +1436,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # COOKIE_FILE if it is not set.  With a non-interactive Lynx session (e.g.,
 # -dump), cookies will only be saved to file if COOKIE_SAVE_FILE is set.
 #
-#COOKIE_SAVE_FILE:~/.lynx_cookies
+#COOKIE_SAVE_FILE:~/.etc/cookies.txt
 
 
 .h1 Mail-related
@@ -1579,7 +1577,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # It will be prepended by the user's home directory.
 # NOTE that a file ending in .html or other suffix mapped to text/html
 # should be used to ensure its treatment as HTML.  The built-in default
-# is lynx_bookmarks.html.  On both Unix and VMS, if a subdirectory off of
+# is ./.etc/bookmark.htm.  On both Unix and VMS, if a subdirectory off of
 # the HOME directory is desired, the path should begin with "./" (e.g.,
 # ./BM/lynx_bookmarks.html), but the subdirectory must already exist.
 # Lynx will create the bookmark file, if it does not already exist, on
@@ -1591,7 +1589,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # (see below), via the 'o'ptions menu, and can save those definitions
 # in the .lynxrc file.
 #
-#DEFAULT_BOOKMARK_FILE:lynx_bookmarks.html
+#DEFAULT_BOOKMARK_FILE:./.etc/bookmark.htm
 
 
 .h2 MULTI_BOOKMARK_SUPPORT
@@ -1674,6 +1672,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 #POSITIONABLE_EDITOR:jove
 #POSITIONABLE_EDITOR:jpico
 #POSITIONABLE_EDITOR:jstar
+#POSITIONABLE_EDITOR:jupp
 #POSITIONABLE_EDITOR:nano
 #POSITIONABLE_EDITOR:pico
 #POSITIONABLE_EDITOR:rjoe
@@ -1857,7 +1856,11 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 .ex
 #DOWNLOADER:Save OS/390 binary file: iconv -f IBM-1047 -t ISO8859-1 %s >%s:FALSE
 
+# These are configured by The MirOS Project:
+DOWNLOADER:View in less:less -f '%s':FALSE
+DOWNLOADER:Spawn editor:"${VISUAL-${EDITOR-ed}}" '%s':FALSE
 
+
 .h1 Interaction
 
 .h2 NO_DOT_FILES
@@ -1997,7 +2000,7 @@ DEFAULT_INDEX_FILE:http://lynx.isc.org/
 # prefer the more conservative action of requiring an explicit Y or y to
 # confirm.  The default defined here will override that in userdefs.h.
 #
-#QUIT_DEFAULT_YES:TRUE
+QUIT_DEFAULT_YES:FALSE
 
 
 .h1 HTML Parsing
@@ -2335,7 +2338,7 @@ MINIMAL_COMMENTS:TRUE
 # XLOADIMAGE_COMMAND definition.
 # If you use xli, you may want to add the -quiet flag.
 #
-#XLOADIMAGE_COMMAND:xli %s &
+#XLOADIMAGE_COMMAND:xloadimage -quiet %s
 
 .h2 VIEWER
 # MIME types may be assigned to external viewers using
@@ -3292,7 +3295,7 @@ COLOR:6:brightred:black
 # otherwise. Set this setting to TRUE if you prefer numbered links, but wish
 # to get valid HTML source when printing or mailing when in psrc view.
 # Default is FALSE.
-#PRETTYSRC_VIEW_NO_ANCHOR_NUMBERING:FALSE
+PRETTYSRC_VIEW_NO_ANCHOR_NUMBERING:TRUE
 
 
 .h1 HTML Parsing
@@ -3328,19 +3331,17 @@ COLOR:6:brightred:black
 #	<p />
 #	<a />
 # When the option is false, lynx will not treat the tag as an ending.
-#XHTML_PARSING:FALSE
+XHTML_PARSING:TRUE
 
 .h1 Appearance
 
 .h2 JUSTIFY
 # JUSTIFY - Appearance
-# This option mirrors command-line option with same name.  Default is TRUE.  If
-# true, most of text (except headers and like this) will be justified.  This
-# has no influence on CJK text rendering.
+# This option mirrors command-line option with same name.  Default is FALSE.
+# If true, most of text (except headers and like this) will be justified.
+# This has no influence on CJK text rendering.
 #
-# This option is only available if Lynx was compiled with EXP_JUSTIFY_ELTS.
-#
-#JUSTIFY:FALSE
+JUSTIFY:TRUE
 
 .h2 JUSTIFY_MAX_VOID_PERCENT
 # JUSTIFY_MAX_VOID_PERCENT - Appearance
@@ -3368,7 +3369,7 @@ COLOR:6:brightred:black
 # explicit activation can also be requested with the -tna command line
 # option.
 #
-#TEXTFIELDS_NEED_ACTIVATION:FALSE
+TEXTFIELDS_NEED_ACTIVATION:TRUE
 
 .h2 LEFTARROW_IN_TEXTFIELD_PROMPT
 # LEFTARROW_IN_TEXTFIELD_PROMPT
@@ -3378,7 +3379,7 @@ COLOR:6:brightred:black
 # only if the contents of the fields have been changed since entering it.
 # If set to TRUE, the confirmation prompt is always issued.
 #
-#LEFTARROW_IN_TEXTFIELD_PROMPT:FALSE
+LEFTARROW_IN_TEXTFIELD_PROMPT:TRUE
 
 
 .h1 Timeouts
@@ -3419,7 +3420,7 @@ COLOR:6:brightred:black
 # one lynx.cfg setting.
 .nf
 #ENABLE_LYNXRC:accept_all_cookies:ON
-#ENABLE_LYNXRC:assume_charset:OFF
+ENABLE_LYNXRC:assume_charset:ON
 #ENABLE_LYNXRC:auto_session:OFF
 #ENABLE_LYNXRC:bookmark_file:ON
 #ENABLE_LYNXRC:case_sensitive_searching:ON
@@ -3435,37 +3436,37 @@ COLOR:6:brightred:black
 #ENABLE_LYNXRC:emacs_keys:ON
 #ENABLE_LYNXRC:file_editor:ON
 #ENABLE_LYNXRC:file_sorting_method:ON
-#ENABLE_LYNXRC:force_cookie_prompt:OFF
-#ENABLE_LYNXRC:force_ssl_prompt:OFF
-#ENABLE_LYNXRC:ftp_passive:OFF
+ENABLE_LYNXRC:force_cookie_prompt:ON
+ENABLE_LYNXRC:force_ssl_prompt:ON
+ENABLE_LYNXRC:ftp_passive:ON
 #ENABLE_LYNXRC:kblayout:ON
 #ENABLE_LYNXRC:keypad_mode:ON
 #ENABLE_LYNXRC:lineedit_mode:ON
 #ENABLE_LYNXRC:locale_charset:ON
-#ENABLE_LYNXRC:make_links_for_all_images:OFF
-#ENABLE_LYNXRC:make_pseudo_alts_for_inlines:OFF
+ENABLE_LYNXRC:make_links_for_all_images:ON
+ENABLE_LYNXRC:make_pseudo_alts_for_inlines:ON
 #ENABLE_LYNXRC:multi_bookmark:ON
-#ENABLE_LYNXRC:no_pause:OFF
+ENABLE_LYNXRC:no_pause:ON
 #ENABLE_LYNXRC:personal_mail_address:ON
 #ENABLE_LYNXRC:preferred_charset:ON
-#ENABLE_LYNXRC:preferred_encoding:OFF
+ENABLE_LYNXRC:preferred_encoding:ON
 #ENABLE_LYNXRC:preferred_language:ON
-#ENABLE_LYNXRC:preferred_media_types:OFF
-#ENABLE_LYNXRC:raw_mode:OFF
+ENABLE_LYNXRC:preferred_media_types:ON
+ENABLE_LYNXRC:raw_mode:ON
 #ENABLE_LYNXRC:run_all_execution_links:ON
 #ENABLE_LYNXRC:run_execution_links_on_local_files:ON
-#ENABLE_LYNXRC:scrollbar:OFF
+ENABLE_LYNXRC:scrollbar:ON
 #ENABLE_LYNXRC:select_popups:ON
 #ENABLE_LYNXRC:send_useragent:OFF
 #ENABLE_LYNXRC:session_file:OFF
-#ENABLE_LYNXRC:set_cookies:OFF
+ENABLE_LYNXRC:set_cookies:ON
 #ENABLE_LYNXRC:show_color:ON
 #ENABLE_LYNXRC:show_cursor:ON
 #ENABLE_LYNXRC:show_dotfiles:ON
-#ENABLE_LYNXRC:show_kb_rate:OFF
+ENABLE_LYNXRC:show_kb_rate:ON
 #ENABLE_LYNXRC:sub_bookmarks:ON
-#ENABLE_LYNXRC:tagsoup:OFF
-#ENABLE_LYNXRC:underline_links:OFF
+ENABLE_LYNXRC:tagsoup:ON
+ENABLE_LYNXRC:underline_links:ON
 #ENABLE_LYNXRC:user_mode:ON
 #ENABLE_LYNXRC:useragent:OFF
 #ENABLE_LYNXRC:verbose_images:ON
